itou/utils/htmx/tests.py

```diff
@@ -0,0 +1,37 @@
+from django.http import HttpResponse
+from django.test import SimpleTestCase
+from django_htmx.middleware import HtmxMiddleware
+
+from .utilities_for_testing import HtmxRequestFactory
+
+
+class HtmxRequestFactoryTest(SimpleTestCase):
+    def setUp(self):
+        self.htmx_request = HtmxRequestFactory()
+        self.middleware = HtmxMiddleware(HttpResponse)
+
+    def test_get(self):
+        request = self.htmx_request.get("/")
+        response = self.middleware(request)
+        assert response.status_code == 200
+        assert bool(request.htmx) is True
```

Comment:
Je ne comprends pas ce qu’on vérifie ? `hasattr(request, "htmx")` ? (même remarque plus bas)

Reply:
Pourquoi pas `assert isinstance(request.htmx, HtmxDetails)`, pour vérifier que l’objet `request.htmx` est bien ce que l’on attend ?
En fait, la lecture de `assert bool(request.htmx) is True` se lit comme “évaluer request.htmx comme un bool doit retourner True”. Maintenant que je comprends mieux, j’imagine que c’est pour pouvoir écrire `if request.htmx:` dans la vue, afin de tester si la requête a été faite avec htmx. Ce n’était pas évident pour moi, mais c’est peut-être juste mon problème.
Comme tu préfères.

Reply:
Ce sera en effet plus lisible. Je change.